lib/cli/src/commands/app/deploy.rs

```diff
@@ -279,10 +279,40 @@ impl AsyncCliCommand for CmdAppDeploy {
             None
         };
 
-        let owner = self
+        let mut owner = self
             .get_owner(&client, &mut app_yaml, maybe_edge_app.as_ref())
             .await?;
 
+        if !wasmer_api::query::viewer_can_deploy_to_namespace(&client, &owner).await? {
```

Comment:
This is not the way. We can simply fetch that the app exists (given the app_id if existing, or using the namespace/name), and if not... try to deploy it as normal and let it fail if the backend fails

Reply:
FTR: We agreed that just checking if the user can deploy to a namespace is simpler than fetching the app, as receiving just `None` from the backend does not tell us which situation the user is in: do they have permissions to deploy an app and it does not exist (yet) or the user can't deploy to the selected namespace and should thus change the owner?